lighthouse-core/audits/bootup-time.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+const DevtoolsTimelineModel = require('../lib/traces/devtools-timeline-model');
+const Util = require('../report/v2/renderer/util.js');
+
+class BootupTime extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'bootup-time',
+      description: 'JavaScript boot-up time is high (> 4s)',
+      failureDescription: 'JavaScript boot-up time is too high.',
+      helpText: 'Consider reducing the time spent parsing, compiling and executing JS. ' +
+        'You may find delivering smaller JS payloads helps with this.',
+      requiredArtifacts: ['traces'],
+    };
+  }
+
+  /**
+   * @param {!Array<TraceEvent>=} trace
+   * @return {!Map<string, Number>}
+   */
+  static getExecutionTimingsByURL(trace) {
+    const timelineModel = new DevtoolsTimelineModel(trace);
```

Comment:
are we doing this in a lot of places/should it be a computed artifact?

Reply:
Are you referring to make a computed artifact `requestExecutionTimingsByUrl(trace)` or rather have a computed artifact `requestExecutionTimings(trace, 'URL')`

Reply:
the idea is to make DTM a computed artifact.

we were also discussing doing a deepFreeze (from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) on the trace data before sending it into devtools.

so it wont mutate.

Reply:
should i do it in this PR? or just a followup?

Reply:
filed here https://github.com/GoogleChrome/lighthouse/issues/3702